src/components/board/client.ts

```diff
@@ -202,6 +203,32 @@ export class BoardClient implements Board {
     return response.getValue();
   }
 
+  async setPowerMode(
+    name: string,
+    powerMode: PowerMode,
+    durationObj?: DurationOptions,
+    extra = {}
+  ) {
+    const { boardService } = this;
+    const request = new pb.SetPowerModeRequest();
+    request.setName(name);
+    request.setPowerMode(powerMode);
+    if (durationObj) {
+      const duration = new Duration();
+      duration.setNanos(durationObj.nanos);
+      duration.setSeconds(durationObj.seconds);
+      request.setDuration(duration);
+    }
```

Comment:
[minor] I'd prefer if we named the user-facing variables and types as `duration`.

```suggestion
  async setPowerMode(
    name: string,
    powerMode: PowerMode,
    duration?: Duration,
    extra = {}
  ) {
    const { boardService } = this;
    const request = new pb.SetPowerModeRequest();
    request.setName(name);
    request.setPowerMode(powerMode);
    if (duration) {
      const pbDuration = new PBDuration();
      duration.setNanos(duration.nanos);
      duration.setSeconds(duration.seconds);
      request.setDuration(pbDuration);
    }
```